core/src/immutable_deserialized_packet.rs

```diff
@@ -129,8 +129,12 @@ impl PartialOrd for ImmutableDeserializedPacket {
 }
 
 impl Ord for ImmutableDeserializedPacket {
+    /// Order by priority (higher first), then by compute_unit_limit (lower first)
     fn cmp(&self, other: &Self) -> Ordering {
-        self.priority().cmp(&other.priority())
+        match self.priority().cmp(&other.priority()) {
+            Ordering::Equal => other.compute_unit_limit().cmp(&self.compute_unit_limit()),
```

Comment:
Not fully convinced if it's a good idea to prioritize txs requested less CUs. Maybe better to incentivize user to adapt requested CU with economic means. But if we really want to using prioritization as tool to encourage user to requested CU, then maybe only by a boolean flag of requested vs non-requested. (perhaps better move the discussion to issue itself)

Reply:
Does reversing the comparison result in better block packing?